example/01_gemm/gemm_xdl_input_i32_comp_i8_scale_ab.cpp

```diff
@@ -0,0 +1,301 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2018-2023, Advanced Micro Devices, Inc. All rights reserved.
+
+#include <iostream>
+#include <numeric>
+#include <initializer_list>
+#include <cstdlib>
+
+#include "ck/ck.hpp"
+#include "ck/tensor_operation/gpu/device/gemm_specialization.hpp"
+#include "ck/tensor_operation/gpu/device/impl/device_gemm_multiple_d_scale_ab_xdl_cshuffle.hpp"
+#include "ck/tensor_operation/gpu/element/element_wise_operation.hpp"
+
+#include "ck/library/utility/device_memory.hpp"
+#include "ck/library/utility/host_tensor.hpp"
+#include "ck/library/utility/host_tensor_generator.hpp"
+#include "ck/library/utility/literals.hpp"
+#include "ck/library/reference_tensor_operation/cpu/reference_gemm.hpp"
+#include "ck/library/utility/check_err.hpp"
+
+template <ck::index_t... Is>
+using S = ck::Sequence<Is...>;
+
+using I8  = int8_t;
+using I32 = int32_t;
+
+using Row = ck::tensor_layout::gemm::RowMajor;
+using Col = ck::tensor_layout::gemm::ColumnMajor;
+
+using PassThrough = ck::tensor_operation::element_wise::PassThrough;
+
+using ADataType        = I32;
+using BDataType        = I32;
+using AccDataType      = I32;
+using CShuffleDataType = I32;
+using EDataType        = I32;
+using ComputeType      = I8;
+
+using ALayout = Row;
+using BLayout = Col;
+using ELayout = Row;
+
+struct i32_to_i8
+{
+    __host__ __device__ void operator()(I8& y, const I32& x) const
+    {
+        float scale = (1.0 / reduced_amax) * int8_max;
+        y           = ck::type_convert<I8>(ck::type_convert<float>(x) * scale);
+    }
+
+    static constexpr float int8_max = 127;
+    // this attribute will trigger a reduction op of the tensor to get the true amax scalue
+    float reduced_amax = 1.0;
+};
+
+struct i8_to_i32
+{
+    __host__ __device__ void operator()(I32& y, const I8& x) const
+    {
+        float a_scale = (1.0 / a_reduced_amax) * int8_max;
+        float b_scale = (1.0 / b_reduced_amax) * int8_max;
+        float c_scale = (1.0 / (a_scale * b_scale));
+
+        y = ck::type_convert<I32>(ck::type_convert<float>(x) * c_scale);
+    }
+
```

Comment:
@rocking5566 can you take a look whether this is correct ?

Reply:
1. Just curious, I wonder is it int8 dynamic quantization?
2. Why A, B, C type are i32? The calculation seem like float to i8?
3. Use clip operation instead of type_convert when narrowing down the type (eg. i32 to i8). This might be different.
4. Does the goal of CDEElementOp is to output the original range of tensor? Because I do not see the c_reduced_amax
5. CDEElementOp could be simplify into (a_reduced_amax * b_reduced_amax) / (int8_max * int8_max) to reduce the calculation